eng/native/init-compiler.sh

```diff
@@ -106,6 +107,13 @@ if [[ -z "$CC" ]]; then
     exit 1
 fi
 
+if command -v "lld$desired_version" > /dev/null; then
+    # Only lld version >= 9 can be considered stable
+    if [[ "$majorVersion" -ge 9 ]]; then
```

Comment:
a nit:
```suggestion
    if [[ "$compiler" == "clang" && "$majorVersion" -ge 9 ]]; then
```

Reply:
Done. I didn't originally want to make LLD usage clang specific, since gcc can use it too, but it is true that using gcc version for lld doesn't make sense.